### PR TITLE
feat: generate signed URLs for blobs/buckets using virtual hostname

### LINKED ISSUE #10: [Storage] V4 Signature Support for virtual hosted-style signed
Tracking issue for work defined in companion document.

[New Client Request - Support for virtual hosted-style signed URLs](https://docs.google.com/document/d/1TkoDiZjz6nINNVQrYbrfaODhmcMTR-crb94aR1cXZVA/edit#)

### PR DESCRIPTION
**Pull request by tseaver**
Closes #10.

### CODE REVIEW

File: google/cloud/storage/blob.py [hunk 1]
```diff
@@ -361,6 +361,7 @@ def generate_signed_url(
         version=None,
         service_account_email=None,
         access_token=None,
+        use_virtual_hostname=False,
```
Comment (frankyn) [inline google/cloud/storage/blob.py]: Please use `virtual_hosted_style` instead.

Comment (frankyn) [inline google/cloud/storage/blob.py]: Please use `virtual_host_style`

Comment (tseaver) [reply google/cloud/storage/blob.py]: Which spelling?  You typed `virtual_hosted_style` above. :)

Comment (tseaver) [reply google/cloud/storage/blob.py]: I picked this one, as it is shorter. LMK if I guessed wrong.

Comment (frankyn) [reply google/cloud/storage/blob.py]: Whoops :man_facepalming: use `virtual_hosted_style` so it's consistent across languages.

Comment (tseaver) [reply google/cloud/storage/blob.py]: Done: b75fd1d

**Pull request merged**